### DISCUSSION TITLE
Handwritten annotations

### DISCUSSION THREAD

**Discussion by zuluwi** [RyotaUshio/obsidian-pdf-plus#42] [Q&A]
When working with PDFs, PDF++ alone can often provide the necessary workflow. Since I often work with workbooks, I need to draw on pdfs, which requires a separate program. 

I would like to ask two things about this
- Is there any plan to add a feature to the plugin for adding drawing annotations to the pdf?
- Is there a workflow you use to work simultaneously with PDF++ and the pdf drawing program on the same pdf?

**Comment by RyotaUshio** [comment]
Hi, sorry for the late reply!

Handwriting is one of the most important open problems about PDFs in Obsidian.
I also often need to handwrite when I read math books for example, and I have been frustrated by how Obsidian forced me quite an awkward workflow when it came to handwriting.

## Is there any plan to add a feature to the plugin for adding drawing annotations to the pdf?

Probably, no. But this is a "positive no".

Obsidian uses PDF.js as the underlying PDF viewer library, and the [roadmap](https://obsidian.md/roadmap/) says

> PDF annotation:
> Currently waiting for native support in PDF.js.

However, PDF.js already supports ink annotation. You can try it in Firefox or their Chrome extension:

https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/deb56d58-b3b4-4a62-b3ac-d36a09d371c2

So when Obsidian updates their PDF viewer the next time, ink annotation _might_ be included as a core feature.
Given this situation, probably I will not add handwriting functionality to PDF++.

By the way, I found this on Reddit: https://www.reddit.com/r/ObsidianMD/comments/185xzmq/pdf_annotations_soon_progress_on_github_page_of/

## Is there a workflow you use to work simultaneously with PDF++ and the pdf drawing program on the same pdf?

I'm not 100% satisfied with my current workflow, but here I'll explain how I do it for now.

First of all, I primarily use my desktop (MacBook Pro), and I only use my iPad when I need to do handwriting.
I use Obsidian Sync to synchronize files between devices.
I'm sorry but I'm not familiar with other platforms.

There are a bunch of iPad apps that _seem to_ let me handwrite on PDFs. 
But many of them, including GoodNotes, force me to "import" PDFs into their apps in some other proprietary format.
They let me handwrite, but actually, I cannot draw on the original PDF file itself. When I need my handwriting in PDF format, I need to _export_ it as a new PDF document.

This makes it difficult to integrate those apps with Obsidian. In other words, handwriting should be added to the original PDF file itself.

### Workflow on iPad

... I talked too much. My current approach to accomplish the above goal is the following:

1. On your iPad, open the PDF file that you want to handwrite on.
2. Tap the "more options" (three dots) button and select "Share" (or, just run the command "Share").
3. Select [Markup](https://support.apple.com/en-us/HT206885). 
    <img width="300" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/40f1b6a8-5ed2-46e8-9919-c2e8456796eb">
4. Now you can draw on the PDF.    
    <img width="500" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/badec2a8-43f0-4ffa-b1ac-b417a45930a8">
5. When you finish drawing, tap "Done" and then "Delete PDF". (This part is tricky; selecting "delete PDF" does not actually delete the original PDF. I'm not sure why Apple designed it in such as way.)
6. Now your drawing or handwriting appears in Obsidian. Thanks to Obsidian Sync (or any other sync service you choose), it will be also visible on your desktop.
    <img width="500" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/984fc238-f80a-4b4e-90cc-f9f9ce945fa0">

> [!tip]
> Step 1 & 2 can be a bit tedious, but today I released a new plugin called [View Sync](https://github.com/RyotaUshio/obsidian-view-sync), which eliminates these steps by allowing Obsidian on iPad to automatically open the PDF file that I open on my desktop.

> [!tip]
> The [handwritten notes](https://github.com/FBarrca/obsidian-handwritten-notes) plugin adds a button that lets you go through Step 2 just by tapping that button. The above workflow is essentially the same as the one that the plugin proposes.

> [!note]
> In step 3, Markup can be replaced with any other tools that let you add drawings directly to the original PDF file.
> - NotesWriter Pro
> - Xodo
> - PDF Viewer

**Reply by Dennyyusuf** [reply]
> Hi, sorry for the late reply!
> 
> Handwriting is one of the most important open problems about PDFs in Obsidian. I also often need to handwrite when I read math books for example, and I have been frustrated by how Obsidian forced me quite an awkward workflow when it came to handwriting.
> 
> ## Is there any plan to add a feature to the plugin for adding drawing annotations to the pdf?
> Probably, no. But this is a "positive no".
> 
> Obsidian uses PDF.js as the underlying PDF viewer library, and the [roadmap](https://obsidian.md/roadmap/) says
> 
> > PDF annotation:
> > Currently waiting for native support in PDF.js.
> 
> However, PDF.js already supports ink annotation. You can try it in Firefox or their Chrome extension:
> 
>  Screen.Recording.2024-02-11.at.19.41.16.mov 
> 
> So when Obsidian updates their PDF viewer the next time, ink annotation _might_ be included as a core feature. Given this situation, probably I will not add handwriting functionality to PDF++.
> 
> By the way, I found this on Reddit: https://www.reddit.com/r/ObsidianMD/comments/185xzmq/pdf_annotations_soon_progress_on_github_page_of/
> 
> ## Is there a workflow you use to work simultaneously with PDF++ and the pdf drawing program on the same pdf?
> I'm not 100% satisfied with my current workflow, but here I'll explain how I do it for now.
> 
> First of all, I primarily use my desktop (MacBook Pro), and I only use my iPad when I need to do handwriting. I use Obsidian Sync to synchronize files between devices. I'm sorry but I'm not familiar with other platforms.
> 
> There are a bunch of iPad apps that _seem to_ let me handwrite on PDFs. But many of them, including GoodNotes, force me to "import" PDFs into their apps in some other proprietary format. They let me handwrite, but actually, I cannot draw on the original PDF file itself. When I need my handwriting in PDF format, I need to _export_ it as a new PDF document.
> 
> This makes it difficult to integrate those apps with Obsidian. In other words, handwriting should be added to the original PDF file itself.
> 
> ### Workflow on iPad
> ... I talked too much. My current approach to accomplish the above goal is the following:
> 
> 1. On your iPad, open the PDF file that you want to handwrite on.
> 2. Tap the "more options" (three dots) button and select "Share" (or, just run the command "Share").
> 3. Select [Markup](https://support.apple.com/en-us/HT206885).
>    <img alt="image" width="300" src="https://private-user-images.githubusercontent.com/72342591/303913889-40f1b6a8-5ed2-46e8-9919-c2e8456796eb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyNTU3NTMsIm5iZiI6MTc1NjI1NTQ1MywicGF0aCI6Ii83MjM0MjU5MS8zMDM5MTM4ODktNDBmMWI2YTgtNWVkMi00NmU4LTk5MTktYzJlODQ1Njc5NmViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI3VDAwNDQxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1YTU5ZGVmMTZkMDBkZTZiNDgyOGY3YWQ2MDU4N2Q5MWYwOWJmNWRjZWVjZTdmNDY1YjA3YWRkNzM1NGM3NjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.i8VzFnovyMgmX184uniEfS_DHStJH0TJtEkcUn8Jsu0">
> 4. Now you can draw on the PDF.
>    <img alt="image" width="500" src="https://private-user-images.githubusercontent.com/72342591/303913950-badec2a8-43f0-4ffa-b1ac-b417a45930a8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyNTU3NTMsIm5iZiI6MTc1NjI1NTQ1MywicGF0aCI6Ii83MjM0MjU5MS8zMDM5MTM5NTAtYmFkZWMyYTgtNDNmMC00ZmZhLWIxYWMtYjQxN2E0NTkzMGE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI3VDAwNDQxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2MTQyNmU5ZDM1Mjc4ZTEwNmM0MjZmNjA0MzczNWM5ODBlZmU2Y2VhZDE1NjU0NTQ0MjU0YTA1OTc2MTc1YjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.N1303U7cbIsqYXl-oUBgIzoeaW9RCxP29d8ViByekt4">
> 5. When you finish drawing, tap "Done" and then "Delete PDF". (This part is tricky; selecting "delete PDF" does not actually delete the original PDF. I'm not sure why Apple designed it in such as way.)
> 6. Now your drawing or handwriting appears in Obsidian. Thanks to Obsidian Sync (or any other sync service you choose), it will be also visible on your desktop.
>    <img alt="image" width="500" src="https://private-user-images.githubusercontent.com/72342591/303914058-984fc238-f80a-4b4e-90cc-f9f9ce945fa0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyNTU3NTMsIm5iZiI6MTc1NjI1NTQ1MywicGF0aCI6Ii83MjM0MjU5MS8zMDM5MTQwNTgtOTg0ZmMyMzgtZjgwYS00YjRlLTkwY2MtZjlmOWNlOTQ1ZmEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI3VDAwNDQxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3YTE3MTlmNDc4YzE1NzJlYTc2MmFlMDU2NjM1YTNmMWM5MmIwMmE3MGE3ZTgyOWEwYmY1ZDJiOTZhZTJlNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jRKogWN4yNCjtWBLHkLXIsWsjsYWJ5E5nFP4NQN5KsA">
> 
> Tip
> 
> Step 1 & 2 can be a bit tedious, but today I released a new plugin called [View Sync](https://github.com/RyotaUshio/obsidian-view-sync), which eliminates these steps by allowing Obsidian on iPad to automatically open the PDF file that I open on my desktop.
> Tip
> 
> The [handwritten notes](https://github.com/FBarrca/obsidian-handwritten-notes) plugin adds a button that lets you go through Step 2 just by tapping that button. The above workflow is essentially the same as the one that the plugin proposes.
> Note
> 
> In step 3, Markup can be replaced with any other tools that let you add drawings directly to the original PDF file.
> 
> * NotesWriter Pro
> * Xodo
> * PDF Viewer

Hi 

When marking, the software doesn’t automatically jump to the corresponding view, even if that view has already been synced in Obsidian. Instead, when I open the PDF for marking through the sharing menu, I still have to manually locate the correct page.

**Reply by RyotaUshio** [reply]
I'm aware of that but Apple's Markup tool is obviously out of my control.

**Reply by Dennyyusuf** [reply]
Thank you for your reply! I had misunderstood and thought it was just my issue. Are there any other PDF tools for iPad that could handle this linked to the corresponding page?

**Reply by RyotaUshio** [reply]
Not that I'm aware of.

**Comment by JacobH140** [comment]
For situations where I want to be able to simulate 'annotating in the margins' of the PDF, I have gotten some mileage out of the [Obsidian Ink](https://github.com/daledesilva/obsidian_ink) plugin:
![CleanShot 2025-06-19 at 17 57 23@2x](https://github.com/user-attachments/assets/a6b12ff4-c6cd-4292-a17e-e8a34f37e476)
Although this lets one 'do everything without leaving Obsidian', there is a good deal of operational friction. I do find, however, that I have begun preferring this to the mental friction that comes with managing other apps at the same time as Obsidian. Obviously nothing would beat the possibility of directly annotating in Obsidian's PDF viewer.
